### PR TITLE
[flow] Cleanup identifier filter code

### PR DESCRIPTION
**Pull request by smnzhu**
Cleanup some filtering code in identifier and network. Will use this later for IdentityProvider.

### CODE REVIEW

**Review by vishalchangrani — APPROVED** [review]

File: network/p2p/network.go [hunk 1]
```diff
@@ -355,40 +361,14 @@ func (n *Network) multicast(channel network.Channel, message interface{}, num ui
...
-func sampleFilter(size uint) identifierFilter {
-	return func(ids ...flow.Identifier) ([]flow.Identifier, error) {
-		return flow.Sample(size, ids...), nil
+func (n *Network) removeSelfFilter() flow.IdentifierFilter {
+	return func(id flow.Identifier) bool {
+		return id != n.me.NodeID()
 	}
 }
 
 // sendOnChannel sends the message on channel to targets after applying the all the filters to targets.
```
Comment (huitseeker) [inline network/p2p/network.go:370]: Nit: you probably want to correct the comment

**Review by huitseeker — APPROVED** [review]
LGTM!

**Comment by smnzhu** [comment]
bors merge

**Comment by smnzhu** [comment]
bors merge

**Pull request merged**